indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -337,4 +310,44 @@ private Response asLeaderWithSupervisorManager(Function<SupervisorManager, Respo
       return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
     }
   }
+
+  private Set<String> filterAuthorizedSupervisorIds(
+      final HttpServletRequest req,
+      SupervisorManager manager,
+      Collection<String> supervisorIds
+  )
+  {
+    final Set<String> authorizedSupervisorIds = Sets.newHashSet();
+
+    // If there were no supervisorIds, go ahead and authorize the request.
+    if (supervisorIds.size() == 0) {
```

Comment:
Is it necessary to have a separate branch for this case? Shouldn't filterAuthorizedResources on an empty list do the same thing?

Reply:
It's not, I had it there just to avoid creating the resource generating function if there were on supervisorIds, removed this